Model/ProductExport/Images.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Copyright © PureClarity. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+namespace Pureclarity\Core\Model\ProductExport;
+
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\BlockFactory;
+use Magento\Store\Model\Store;
+use Magento\Catalog\Model\Product\Gallery\ReadHandler;
+use Magento\Catalog\Block\Product\View\Gallery;
+use Magento\Catalog\Model\Product;
+use Pureclarity\Core\Model\CoreConfig;
+
+/**
+ * Class Images
+ *
+ * PureClarity Product Export - Images Handler
+ * Gets images for the given product
+ */
+class Images
+{
+    /** @var Gallery $galleryBlock */
+    private $galleryBlock;
+
+    /** @var BlockFactory $blockFactory */
+    private $blockFactory;
+
+    /** @var CoreConfig $coreConfig */
+    private $coreConfig;
+
+    /** @var ReadHandler $galleryReadHandler */
+    private $galleryReadHandler;
+
+    /**
+     * @param BlockFactory $blockFactory
+     * @param CoreConfig $coreConfig
+     * @param ReadHandler $galleryReadHandler
+     */
+    public function __construct(
+        BlockFactory $blockFactory,
+        CoreConfig $coreConfig,
+        ReadHandler $galleryReadHandler
+    ) {
+        $this->blockFactory       = $blockFactory;
+        $this->coreConfig         = $coreConfig;
+        $this->galleryReadHandler = $galleryReadHandler;
+    }
+
+    /**
+     * Gets the main image for a product
+     *
+     * @param Product $product
+     * @param Store $store
+     *
+     * @return string
+     */
+    public function getProductImageUrl($product, $store)
+    {
+        $galleryBlock = $this->getGalleryBlock();
+        $galleryBlock->setData('product', $product);
+            
+        if ($product->getImage() && $product->getImage() !== 'no_selection') {
+            $productImage = $galleryBlock->getImage($product, 'category_page_grid');
+            $productImageUrl = $productImage->getImageUrl();
+        } else {
+            $productImageUrl = $this->coreConfig->getProductPlaceholderUrl($store->getId());
+            if ($productImageUrl === null) {
+                $productImageUrl = $store->getBaseUrl(UrlInterface::URL_TYPE_MEDIA)
+                                 . 'catalog/product/placeholder/'
+                                 . $store->getConfig('catalog/placeholder/image_placeholder');
+            }
+        }
+
+        return $productImageUrl;
+    }
+
+    /**
+     * Gets the gallery images for a product
+     *
+     * @param Product $product
+     *
+     * @return string[]
+     */
+    public function getProductGalleryUrls($product)
+    {
+        $galleryBlock = $this->getGalleryBlock();
+        $galleryBlock->setData('product', $product);
+
+        $this->galleryReadHandler->execute($product);
+        $productImages = $galleryBlock->getGalleryImages()->getItems();
+
+        $allImages = [];
+
+        foreach ($productImages as $image) {
+            $allImages[] = str_replace(
```

Comment:
do we potentially need to handle different cases? i.e. HTTP:// (given we're talking about arbitrary data)

Reply:
Don't think we have to :), these URLs will only come from configuration and Magento doesn't allow you to put "HTTP://" as it treats that as an invalid URL. So in theory, we should never get an uppercase protocol in these URLs.